x-pack/plugins/observability_solution/infra/public/components/asset_details/tabs/dashboards/context_menu.tsx

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { i18n } from '@kbn/i18n';
+
+import React, { useState } from 'react';
+import { EuiButtonIcon, EuiContextMenuPanel, EuiContextMenuItem, EuiPopover } from '@elastic/eui';
+
+interface Props {
+  items: React.ReactNode[];
+}
+
+export function ContextMenu({ items }: Props) {
+  const [isPopoverOpen, setPopover] = useState(false);
+
+  const onButtonClick = () => {
+    setPopover(!isPopoverOpen);
```

Comment:
```suggestion
    setPopover(state => !state);
```

Reply:
I used `useBoolean` hook for the `isPopoverOpen` state, I think it makes sense :)